src/environmentd/tests/sql.rs

```diff
@@ -3366,34 +3365,29 @@ async fn test_retain_history() {
             .await
             .unwrap();
 
-        // With an index the AS OF query should fail because we haven't taught the planner about retain
-        // history yet.
-        client
-            .batch_execute(&format!("CREATE DEFAULT INDEX ON {name}"))
-            .await
-            .unwrap();
-
+        // We should be able to query the views now and in the past.
         let ts = get_explain_timestamp(name, &client).await;
-        assert_contains!(
-            client
-                .query(&format!("SELECT * FROM {name} AS OF {ts}-2000"), &[])
-                .await
-                .unwrap_err()
-                .to_string(),
-            "not valid for all inputs"
-        );
```

Comment:
While this behavior is not optimal, it still seems worthwhile to assert? This way were exercising known behavior?

It's fine to leave as-is, but I'm curious if this assert was failing?

Reply:
This assert was designed to fail due to landing on the index. But we could add a similar assertion to verify that times totally out of bounds of the collection fail.